go/vt/tabletserver/fakesqldb/conn.go

```diff
@@ -0,0 +1,182 @@
+// that should return exactly 1 row.
+func (conn *Conn) ExecuteFetchMap(query string) (map[string]string, error) {
+	qr, err := conn.ExecuteFetch(query, 1, true)
+	if err != nil {
+		return nil, err
+	}
+	if len(qr.Rows) != 1 {
+		return nil, fmt.Errorf("query %#v returned %d rows, expected 1", query, len(qr.Rows))
+	}
+	if len(qr.Fields) != len(qr.Rows[0]) {
+		return nil, fmt.Errorf("query %#v returned %d column names, expected %d", query, len(qr.Fields), len(qr.Rows[0]))
+	}
+
+	rowMap := make(map[string]string)
+	for i, value := range qr.Rows[0] {
+		rowMap[qr.Fields[i].Name] = value.String()
+	}
+	return rowMap, nil
+}
+
+// ExecuteStreamFetch starts a streaming query to db server. Use FetchNext
+// on the Connection until it returns nil or error
+func (conn *Conn) ExecuteStreamFetch(query string) error {
+	if conn.IsClosed() {
+		return fmt.Errorf("Connection is closed")
+	}
+	result, ok := conn.queryMap[query]
+	if !ok {
+		return fmt.Errorf("unsupported query: %s", query)
+	}
+	conn.curQueryResult = result
+	conn.curQueryRow = 0
+	return nil
+}
+
+// Close closes the db connection
+func (conn *Conn) Close() {
+	conn.isClosed = true
+}
+
+// IsClosed returns if the connection was ever closed
+func (conn *Conn) IsClosed() bool {
+	return conn.isClosed
+}
+
+// CloseResult finishes the result set
+func (conn *Conn) CloseResult() {
+}
+
+// Shutdown invokes the low-level shutdown call on the socket associated with
+// a connection to stop ongoing communication.
+func (conn *Conn) Shutdown() {
+}
+
+// Fields returns the current fields description for the query
+func (conn *Conn) Fields() []proto.Field {
+	return make([]proto.Field, 0)
+}
+
+// ID returns the connection id.
+func (conn *Conn) ID() int64 {
+	return conn.id
+}
+
+// FetchNext returns the next row for a query
+func (conn *Conn) FetchNext() ([]sqltypes.Value, error) {
+	rowCount := int64(len(conn.curQueryResult.Rows))
+	if conn.curQueryResult == nil || rowCount <= conn.curQueryRow {
+		return nil, nil
+	}
+	row := conn.curQueryResult.Rows[conn.curQueryRow]
+	conn.curQueryRow++
+	return row, nil
+}
+
+// ReadPacket reads a raw packet from the connection.
+func (conn *Conn) ReadPacket() ([]byte, error) {
+	return []byte{}, nil
+}
+
+// SendCommand sends a raw command to the db server.
+func (conn *Conn) SendCommand(command uint32, data []byte) error {
+	return nil
+}
+
+// GetCharset returns the current numerical values of the per-session character
+// set variables.
+func (conn *Conn) GetCharset() (cs proto.Charset, err error) {
+	return *conn.charset, nil
+}
+
+// SetCharset changes the per-session character set variables.
+func (conn *Conn) SetCharset(cs proto.Charset) error {
+	*conn.charset = cs
+	return nil
+}
+
+// Register registers a fake implementation of sqldb.Conn and returns its registered name
+func Register(queryMap map[string]*proto.QueryResult) string {
+	name := fmt.Sprintf("fake-%d", rand.Int63())
```

Comment:
This will always return the same number unless you seed it with something random.